File: telegram/src/types/agent.ts
export interface AgentTurnRequest {
    message: string;
    provider: string;
    user_id: string;
    thread_id: string;
    username?: string;
    conversation_id?: string;
    selected_task_number?: number;
    client_message_id?: string;
    client_nonce?: string;
}

export interface AgentTurnResponse {
    assistant_message: string | null;
    conversation_id: string;
    thread_id: string;
    selected_task_number: number | null;
    allocated_new_thread: boolean;
    blocked: boolean;
    blocked_message: string | null;
}

File: volo_agent/api/schemas/agent.py
from pydantic import BaseModel, ConfigDict, Field


class AgentTurnRequest(BaseModel):
    model_config = ConfigDict(extra="forbid", str_strip_whitespace=True)
    message: str = Field(min_length=1)
    provider: str = Field(min_length=1)
    user_id: str = Field(min_length=1)
    username: str | None = None
    thread_id: str = Field(min_length=1)
    conversation_id: str | None = None
    selected_task_number: int | None = Field(default=None, ge=1)
    client_message_id: str | None = None
    client_nonce: str | None = None


class AgentTurnResponse(BaseModel):
    model_config = ConfigDict(extra="forbid", str_strip_whitespace=True)
    assistant_message: str | None = None
    conversation_id: str = Field(min_length=1)
    thread_id: str = Field(min_length=1)
    selected_task_number: int | None = Field(default=None, ge=1)
    allocated_new_thread: bool = False
    blocked: bool = False
    blocked_message: str | None = None
